### ISSUE TITLE
[Feature]: Teardown development apps

### ISSUE THREAD

**Issue by stephen-christie-ss** [Shopify/cli#3556]
### What area(s) will this request affect?

App

### What type of change do you want to see?

New feature

### Overview

There is currently no way to programmatically teardown apps.

Proposal: Add a new action, i.e. `shopify app delete -y` to automatically remove apps. This could be limited to non-published "development apps" that are only in use on development shops.

### Motivation

We use GitLab CI for testing and deployment, and follow their recommendation to automatically deploy each feature development branch for review. This means setting up a new app for each development. Once the development is complete, the app is no longer needed. Once a change has passed our review, we no longer need the review app, and have to clean it up later manually.

**Comment by shauns** [comment]
Thank you for sharing this @stephen-christie-ss -- we don't currently support this sort of action for apps via the CLI but its one we'll keep on the radar. I've shared this internally and closing for now.
